src/main/java/com/javadeobfuscator/deobfuscator/transformers/allatori/StringEncryptionTransformer.java

```diff
@@ -85,114 +110,174 @@ public boolean canCheckEquality(JavaValue first, JavaValue second, Context conte
-               invokeCount.get("charAt").get() >= 2 &&
-               invokeCount.get("length").get() >= 1;
+    public static boolean isAllatoriMethod(MethodNode decryptorNode) {
+        return isAllatoriMethod1(decryptorNode) || isAllatoriMethod2(decryptorNode);
+    }
+
+    private static boolean isAllatoriMethod1(MethodNode decryptorNode) {
+        boolean isAllatori = true;
+        isAllatori = isAllatori && TransformerHelper.containsInvokeVirtual(decryptorNode, "java/lang/String", "charAt", "(I)C");
+        isAllatori = isAllatori && TransformerHelper.containsInvokeVirtual(decryptorNode, "java/lang/String", "length", "()I");
+        isAllatori = isAllatori && TransformerHelper.containsInvokeSpecial(decryptorNode, "java/lang/String", "<init>", null);
+        isAllatori = isAllatori && TransformerHelper.countOccurencesOf(decryptorNode, IXOR) > 2;
+        isAllatori = isAllatori && TransformerHelper.countOccurencesOf(decryptorNode, NEWARRAY) > 0;
+        return isAllatori;
     }
 
-    private void patchMethod(Map<String, AtomicInteger> invokeCount, MethodNode method) {
-        if (invokeCount.containsKey("getStackTrace") && invokeCount.containsKey("getClassName")) {
-            for (AbstractInsnNode insn = method.instructions.getFirst(); insn != null; insn = insn.getNext()) {
-                if (insn.getOpcode() == Opcodes.NEW && (((TypeInsnNode) insn).desc.endsWith("Exception") || ((TypeInsnNode) insn).desc.endsWith("Error"))) {
-                    ((TypeInsnNode) insn).desc = "java/lang/RuntimeException";
-                } else if (insn instanceof MethodInsnNode && (((MethodInsnNode) insn).owner.endsWith("Exception") || ((MethodInsnNode) insn).owner.endsWith("Error"))) {
-                    ((MethodInsnNode) insn).owner = "java/lang/RuntimeException";
+    private static boolean isAllatoriMethod2(MethodNode decryptorNode) {
+        Map<Integer, AtomicInteger> map = TransformerHelper.calcOpcodeOccurenceMap(decryptorNode);
+        AtomicInteger zero = new AtomicInteger();
+        boolean isAllatori = true;
+        isAllatori = isAllatori && map.getOrDefault(CASTORE, zero).get() > 0;
+        isAllatori = isAllatori && map.getOrDefault(NEW, zero).get() > 0;
+        isAllatori = isAllatori && map.getOrDefault(IINC, zero).get() > 0;
+        isAllatori = isAllatori && map.getOrDefault(IXOR, zero).get() > 0;
+        isAllatori = isAllatori && map.getOrDefault(I2C, zero).get() > 0;
+        isAllatori = isAllatori && map.getOrDefault(ARETURN, zero).get() > 0;
+        isAllatori = isAllatori && map.getOrDefault(NEWARRAY, zero).get() == 1;
+        isAllatori = isAllatori && TransformerHelper.containsInvokeVirtual(decryptorNode, "java/lang/String", "charAt", "(I)C");
+        isAllatori = isAllatori && TransformerHelper.containsInvokeVirtual(decryptorNode, "java/lang/String", "length", "()I");
+        isAllatori = isAllatori && TransformerHelper.containsInvokeSpecial(decryptorNode, "java/lang/String", "<init>", null);
```

Comment:
These statements might be better with a function call `boolean and(boolean... booleans)`

Reply:
Such a function call would always do every check and not shortcircuit.

Reply:
But this is still doing every check. `if (!condition) return false;` would be better

Reply:
No its not. Shortcircuit is happening here. It checks isAllatori first, if it is false, it doesn't evaluate the other half, as the result of `&&` is already clear.

Reply:
IMHO, this should be fine as is.